app/gui2/shared/ast/parse.ts

```diff
@@ -88,6 +85,12 @@ class Abstractor {
     const codeStart = whitespaceEnd
     const codeEnd = codeStart + tree.childrenLengthInCodeParsed
     const spanKey = nodeKey(codeStart, codeEnd - codeStart)
+    if (this.substitutes?.has(spanKey)) {
+      // NOTE: Nodes in subtrees obtained from `substitutes` are not added to `nodes` / `tokens / `toRaw`.
+      const node = this.substitutes.get(spanKey)!
+      this.substitutes.delete(spanKey)
```

Comment:
The fact were removing entries from the map is buried quite deep. Please add docs to `abstract` method with any useful information about parameters.

Reply:
Changed the map in the API to a function, so that the caller can inject the side-effectful operation.